tests/test_particle_sets.py

```diff
@@ -277,3 +277,23 @@ def test_pfile_array_remove_all_particles(fieldset, mode, tmpdir, pfile_type, np
         pset.remove(-1)
     pfile.write(pset, 1)
     pfile.write(pset, 2)
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+@pytest.mark.parametrize('test', ['dt', 'endtime'])
+def test_pset_execute_dt_0(fieldset, mode, test, npart=2):
+    def SetLat(particle, fieldset, time, dt):
+        particle.lat = .6
+    pset = ParticleSet(fieldset, pclass=ptype[mode],
+                       lon=np.linspace(0, 1, npart, dtype=np.float32),
+                       lat=np.linspace(1, 0, npart, dtype=np.float32))
+    if test == 'dt':
```

Comment:
If-statement can be removed by passing values for `dt` and `endtime` in pairs in the `@pytest` call. E.g.
```
@pytest.mark.parametrize('endtime', 'dt', [(1., 0.), (0., 1.)])
```
(untested)